src/components/menu-surface/menu-surface.tsx

```diff
@@ -106,7 +112,23 @@ export class MenuSurface {
     }
 
     private handleDocumentClick(event) {
-        if (this.open && !isDescendant(event.target, this.host)) {
+        const elementPath = event.path || [];
+
+        if (!this.open) {
+            return;
+        }
+
+        if (this.allowClicksElement) {
+            const clickedInAllowedElement = elementPath.includes(
```

Comment:
Fixup now done.
I know this row can be put in 121 as well but I think it's easier to follow what happens if we name the result here?